### PR TITLE
feat: build multi arch image

### PR DESCRIPTION
**Pull request by tao12345666333**
Signed-off-by: Jintao Zhang <zhangjintao9020@gmail.com>

### CODE REVIEW

**Comment by Yiyiyimu** [comment]
Hi @tao12345666333 we need to use submodule for non-official (not from github or apache) actions ([ref](https://cwiki.apache.org/confluence/display/BUILDS/GitHub+Actions+status#GitHubActionsstatus-Security) and [example](https://github.com/apache/apisix-dashboard/pull/1795))

File: .github/workflows/apisix_push_docker_hub.yaml [hunk 1]
```diff
@@ -15,10 +15,27 @@ jobs:
...
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      # Login
       - name: Login
-        run: echo ${{ secrets.DOCKER_PASSWORD }} | docker login --username=${{ secrets.DOCKER_USERNAME }} --password-stdin
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}  
+          password: ${{ secrets.DOCKER_PASSWORD }}
```
Comment (gxthrj) [inline .github/workflows/apisix_push_docker_hub.yaml:27]: Can't set this secret in the apache project for now
We have discussed it before. 
https://issues.apache.org/jira/browse/INFRA-21756

Comment (Yiyiyimu) [reply .github/workflows/apisix_push_docker_hub.yaml:27]: Just notice INFRA has replied and they allow us to set those secrets

**Review by tokers — APPROVED** [review]

**Comment by tao12345666333** [comment]
@Yiyiyimu  Addressed. PTAL

File: .github/workflows/apisix_push_docker_hub.yaml (hunk 1 above)
Comment (gxthrj) [reply .github/workflows/apisix_push_docker_hub.yaml:29]: Thanks for reminding. I applied to them to add secrets.

**Comment by gxthrj** [comment]
I applied to them to add secrets. The ASF will help to add secrets. 

ref: https://issues.apache.org/jira/browse/INFRA-21756

BTW, the CI was failed.

File: .github/workflows/apisix_push_docker_hub.yaml [hunk 2]
```diff
@@ -14,11 +14,30 @@ jobs:
...
-        run:
-          make build-on-${{ matrix.platform }}
-          make push-on-${{ matrix.platform }}
+      - name: Build and push
+        id: docker_build
+        uses: ./.github/actions/build-push-action
+        with:
+          context: .
+          file: ./${{ matrix.platform }}/Dockerfile
+          platforms: linux/amd64,linux/arm64
```
Comment (Yiyiyimu) [inline .github/workflows/apisix_push_docker_hub.yaml:41]: Why do we need another amd64 image here

**Review by Yiyiyimu — COMMENTED** [review]
Hi @tao12345666333 could you add another test to make sure the arm image would work as expected, like other tests to add a basic smoke test? Or we'll wait till next release to know if it works 🤣

**Comment by tao12345666333** [comment]
sure.

**Comment by tao12345666333** [comment]
> BTW, the CI was failed.

This task is not related to this PR  https://github.com/apache/apisix-docker/pull/165/checks?check_run_id=2562574181

File: .github/workflows/apisix_push_docker_hub.yaml [hunk 3]
```diff
@@ -14,11 +14,30 @@ jobs:
...
+      # Setup Buildx builder
+      - name: Setup Docker Buildx
+        uses: ./.github/actions/setup-buildx-action
+
+      # Login
       - name: Login
-        run: echo ${{ secrets.DOCKER_PASSWORD }} | docker login --username=${{ secrets.DOCKER_USERNAME }} --password-stdin
+        uses: ./.github/actions/login-action
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
```
Comment (gxthrj) [inline .github/workflows/apisix_push_docker_hub.yaml:28]: As the [infra](https://issues.apache.org/jira/browse/INFRA-21756) says, need to use as below
secrets.DOCKERHUB_USER
secrets.DOCKERHUB_TOKEN

**Review by gxthrj — CHANGES_REQUESTED** [review]

File: .github/workflows/apisix_push_docker_hub.yaml (hunk 3 above)
Comment (tao12345666333) [reply .github/workflows/apisix_push_docker_hub.yaml:28]: ok

**Comment by gxthrj** [comment]
> > BTW, the CI was failed.
> 
> This task is not related to this PR https://github.com/apache/apisix-docker/pull/165/checks?check_run_id=2562574181

ok

**Comment by tao12345666333** [comment]
@Yiyiyimu Will #197 replace this one?

**Comment by Yiyiyimu** [comment]
@tao12345666333 Sorry I totally forgot about this PR 🤣

**Comment by tao12345666333** [comment]
@Yiyiyimu should I close this one?

**Comment by Yiyiyimu** [comment]
> @Yiyiyimu should I close this one?

Good to me

**Pull request closed without merging**